### ISSUE TITLE
linter: react/exhaustive-deps false negative regression

### ISSUE THREAD

**Issue by vemoo** [oxc-project/oxc#11871]
### What version of Oxlint are you using?

1.3.0

### What command did you run?

`oxlint ./src`

### What does your `.oxlintrc.json` config file look like?

```jsonc
{
    "$schema": "./node_modules/oxlint/configuration_schema.json",
    "rules": {
        "react/exhaustive-deps": "warn",
        "no-unused-vars": "off"
    },
    "env": {
        "builtin": true,
        "browser": true
    },
    "ignorePatterns": ["dist/**", "out/**", "node_modules/**"]
}
```

### What happened?

This code:
```jsx
import { useCallback, useEffect } from "react";

function Component({ foo }) {
    const log = useCallback(() => {
        console.log(foo);
    }, [foo]);
    useEffect(() => {
        log();
    }, []);
}
```
doesn't emit any warning.

This is a regression of #9788.

This was fixed on 0.16.9 but on 0.17.0 stopped working.

**Comment by camc314** [comment]
HI, thanks for reporting this issue,

in 0.17.0, we removed react from the default plugin set > https://github.com/oxc-project/oxc/pull/11382

could you add `'react'` to the plugins array and try to repro this again?

**Comment by vemoo** [comment]
That was it. Thank you.

**Comment by camc314** [comment]
awesome, thank you. please keep reporting any issues you find!
